sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+    private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+    private static final String NAME = "name";
+    private static final String VERSION = "version";
+    private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
+    private static final String DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+
+    private final ClientLogger logger = new ClientLogger(TextAnalyticsClientBuilder.class);
+    private final List<HttpPipelinePolicy> policies;
+    private final HttpHeaders headers;
+    private final String clientName;
+    private final String clientVersion;
+
+    private String endpoint;
+    private String subscriptionKey;
+    private TokenCredential tokenCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions;
+    private HttpPipeline httpPipeline;
+    private Configuration configuration;
+    private HttpPipelinePolicy retryPolicy;
+    private TextAnalyticsClientOptions clientOptions;
+    private TextAnalyticsServiceVersion version;
+
+    /**
+     * The constructor with defaults.
+     */
+    public TextAnalyticsClientBuilder() {
+        policies = new ArrayList<>();
+        httpLogOptions = new HttpLogOptions();
+
+        Map<String, String> properties = CoreUtils.getProperties(TEXT_ANALYTICS_PROPERTIES);
+        clientName = properties.getOrDefault(NAME, "UnknownName");
+        clientVersion = properties.getOrDefault(VERSION, "UnknownVersion");
+
+        headers = new HttpHeaders()
+            .put(ECHO_REQUEST_ID_HEADER, "true")
+            .put(CONTENT_TYPE_HEADER, CONTENT_TYPE_HEADER_VALUE)
+            .put(ACCEPT_HEADER, ACCEPT_HEADER_VALUE);
+    }
+
+    /**
+     * TODO (shawn): add javadoc
+     *
+     * @return A TextAnalyticsClient with the options set from the builder.
+     */
+    public TextAnalyticsClient buildClient() {
+        return new TextAnalyticsClient(buildAsyncClient());
+    }
+
+    /**
+     * TODO (shawn): add javadoc
+     *
+     * @return A TextAnalyticsAsyncClient with the options set from the builder.
+     */
+    public TextAnalyticsAsyncClient buildAsyncClient() {
+        // Global Env configuration store
+        Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration().clone() : configuration;
+        // Service Version
+        TextAnalyticsServiceVersion serviceVersion =
+            version != null ? version : TextAnalyticsServiceVersion.getLatest();
+
+        // endpoint cannot be null, which is required in request authentication
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+
+        // Http pipeline is already defined, skip rest of customized pipeline process
+        if (httpPipeline != null) {
+            TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
```

Comment:
This logic here is duplicated in below. I would reverse the logic to do:
```java
if (httpPipeline == null) { 
... do all your stuff like add policies
}

// build client now.
TextAnalyticsClientImpl  testAnalyticsApi = new TextAnalyticsClientImplBuilder()
return ...
```

Reply:
updated